definitions/artifacts/keypair.json

```diff
@@ -0,0 +1,63 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema",
+  "$md": {
+    "access": "public",
+    "name": "keypair"
+  },
+  "type": "object",
+  "title": "Keypair",
+  "description": "Public and Private cryptographic keys commonly used for encryption and authentication",
+  "additionalProperties": false,
+  "required": [
+    "data"
+  ],
+  "properties": {
+    "data": {
+      "type": "object",
+      "required": [
+        "infrastructure",
+        "authentication"
+      ],
+      "properties": {
+        "infrastructure": {
+          "title": "Cloud provider configuration",
+          "description": "Cloud specific Redis configuration data.",
+          "type": "object",
+          "required": [
+            "hostname"
```

Comment:
This seems like another use-case for the `links` block we've been discussing, instead of infrastructure.